backend/src/main/java/co/inventorsoft/scripty/controller/RegistrationController.java

```diff
@@ -0,0 +1,88 @@
+package co.inventorsoft.scripty.controller;
+import co.inventorsoft.scripty.entity.User;
+import co.inventorsoft.scripty.entity.VerificationToken;
+import co.inventorsoft.scripty.entity.dto.UserDto;
+import co.inventorsoft.scripty.registration.RegistrationEvent;
+import co.inventorsoft.scripty.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.web.bind.annotation.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+@RestController
+public class RegistrationController {
+
+    private UserService userService;
+    private ApplicationEventPublisher applicationEventPublisher;
+    private JavaMailSender javaMailSender;
+
+    @Autowired
+    public RegistrationController(UserService userService,
+                                  ApplicationEventPublisher applicationEventPublisher,
+                                  JavaMailSender javaMailSender) {
+        this.userService = userService;
+        this.applicationEventPublisher = applicationEventPublisher;
+        this.javaMailSender = javaMailSender;
+    }
+
+    @PostMapping(value = "/registration", consumes = "application/json")
+    public ResponseEntity registerUserAccount(@Valid @RequestBody UserDto userDto, HttpServletRequest request){
+      if(!(userDto.getPassword()).equals(userDto.getMatchingPassword())){
+          return ResponseEntity
+                  .status(HttpStatus.CONFLICT)
```

Comment:
To indicate validation failure, it's common practice to use BAD_REQUEST(400) status code